src/pint/models/model_builder.py

```diff
@@ -817,3 +915,149 @@ def get_model_and_toas(
         mm.add_tzr_toa(tt)
 
     return mm, tt
+
+
+def guess_binary_model(parfile_dict):
+    """Based on the PINT parameter dictionary, guess the binary model
+
+    Parameters
+    ----------
+    parfile_dict
+        The parameter dictionary as read-in by parse_parfile
+
+    Returns
+    -------
+    list:
+        A priority-ordered list of possible binary models. The first one is the
+        best-guess
+
+    """
+
+    def add_sini(parameters):
+        """If 'KIN' is a model parameter, Tempo2 doesn't really use SINI"""
+        if "KIN" in parameters:
+            return list(parameters) + ["SINI"]
+        else:
+            return list(parameters)
```

Comment:
Naively, why is this needed? Isn't KIN providing the same information as SINI?

Reply:
It's because we need to check parameters against what's in the parfile. PINT doesn't use SINI in DDK, so PINT would find SINI and conclude DDK is not a suitable model. This is just a workaround to make sure PINT would match the DDK model for a T2 model